packages/fx-core/src/component/generator/utils.ts

```diff
@@ -0,0 +1,116 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+import Mustache from "mustache";
+import path from "path";
+import { selectTag } from "../../common/template-utils/templates";
+import { fetchTemplateTagList } from "../../common/template-utils/templatesUtils";
+import {
+  sampleDownloadBaseUrl,
+  sampleTagListUrl,
+  templateDownloadBaseUrl,
+  templateFileExt,
+  templateTagListUrl,
+} from "./constant";
+import {
+  FetchSampleUrlWithTagError,
+  FetchZipFromUrlError,
+  TemplateZipFallbackError,
+  UnzipError,
+} from "./error";
+import { ScaffoldAction, ScaffoldActionName } from "./scaffoldAction";
+import { ScaffoldContext } from "./scaffoldContext";
+
+export async function fetchTemplateUrl(
```

Comment:
This one and the following one looks the same, could we merge into one?

Reply:
done.